### PR TITLE
feat(iteration): UI for create nested iteration (Fix #1316)

### LINKED ISSUE #1316: UI for creating nested iteration
Story #1170

### PR DESCRIPTION
**Pull request by vikram-raj**
Story: #1170

### CODE REVIEW

**Comment by mindreeper2420** [comment]
@vikram-raj I pulled this locally and ran - having a small issue, but not sure if it is just my environment.
When I go to select add parent iteration's name to a new iteration (to make a child iteration), the parent's name just freezes. It then is stuck there for all of the other new iterations I try to make.

screen shot 2017-03-30 at 7 11 07 am

File: src/app/iteration/iteration-modal/iteration-modal.component.html [hunk 1]
```diff
@@ -22,6 +22,18 @@
           </div>
         </div>
         <div class="form-group">
+          <label for="parent-iteration">Parent Iteration</label>
+          <div class="parent-iteration-container">
+            <input id="parent-iteration" class="form-control" type="text" #iterationSearch (keyup)="filterIteration($event)"
```
Comment (mindreeper2420) [inline src/app/iteration/iteration-modal/iteration-modal.component.html]: this should be a typeahead + dropdown (it is only a typeahead when I ran locally).

Comment (vikram-raj) [inline src/app/iteration/iteration-modal/iteration-modal.component.html]: @mindreeper2420 Ok I will make it typeahead + dropdown.

Comment (mindreeper2420) [inline src/app/iteration/iteration-modal/iteration-modal.component.html]: Design for picking a parent iteration:
screen shot 2017-03-30 at 2 09 36 pm

**Comment by vikram-raj** [comment]
@mindreeper2420 I made the changes as you requested. So, can you review it again?

**Comment by mindreeper2420** [comment]
@vikram-raj functionally it works as intended, but that is not the correct dropdown component (style). I'll approve the changes, but it will have to be altered to match how the other dropdowns look/feel (like the Filter dropdowns).

**Pull request merged**